### PR TITLE
Move silicon istype() checks to helper macros, destroy loose code spaghetti

### PR DESCRIPTION
**Pull request by JohnWildkins**
welcome tonight to the latest installment of:

image

no changelog yet because I'm gonna take another look or two at these changes to make sure my regex didn't mess something up

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

**Comment by alsoandanswer** [comment]
Just add a changelog that says "cleans up silicon backends"

File: code/game/objects/items/stacks/medical.dm [hunk 1]
```diff
@@ -22,12 +22,12 @@ Contains:
 	var/apply_sounds
 
 /obj/item/stack/medical/attack(mob/living/M as mob, mob/user as mob)
-	if (!istype(M) || istype(M, /mob/living/silicon) || istype(M, /mob/living/simple_animal/spiderbot))
+	if (!istype(M) || issilicon(M) || istype(M, /mob/living/simple_animal/spiderbot))
 		to_chat(user, "<span class='warning'>\The [src] cannot be applied to [M]!</span>")
 		return 1
 
 	if ( ! (istype(user, /mob/living/carbon/human) || \
-			istype(user, /mob/living/silicon)) )
```
Comment (Arrow768) [inline code/game/objects/items/stacks/medical.dm]: Could put that on the line above/clean up that expression a bit

**Comment by NonQueueingMatt** [comment]
Author requested for this to be closed. Might be best to just remake this entirely with all these conflicts.

**Pull request closed without merging**